### ISSUE TITLE
MethodOverloading false positive if every entry of an enum implement a method.

### ISSUE THREAD

**Issue by jaychang0917** [arturbosch/detekt#2216]
## Expected Behavior
Should not treat enum instance implementation as method overloading.

## Observed Behavior
when every entry of an enum implement a method, it is treated as method overloading.

## Steps to Reproduce
```kotlin
enum class MethodOverloading {
    FOO1 {
        override fun foo() = 1
    },
    FOO2 {
        override fun foo() = 2
    },
    FOO3 {
        override fun foo() = 3
    },
    FOO4 {
        override fun foo() = 4
    },
    FOO5 {
        override fun foo() = 5
    },
    FOO6 {
        override fun foo() = 6
    },
    FOO7 {
        override fun foo() = 7
    };

    abstract fun foo(): Int
}
```

## Your Environment

* Version of detekt used: 1.3.0
* Version of Gradle used (if applicable): 5.4.1

Thanks!

**Comment by schalkms** [comment]
Thanks for reporting this.
